### ISSUE TITLE
[Flatten Array] Impl Class does not make sense

### ISSUE THREAD

**Issue by gabriel376** [exercism/javascript#776]
**Current Implementation**
```Javascript
// flatten-array.spec.js
import { Flattener } from './flatten-array.js';

describe('FlattenArray', () => {
  const flattener = new Flattener();
  test('flattens a nested list', () => {
    expect(flattener.flatten([[]])).toEqual([]);
  });
}

// flatten-array.js
export class Flattener {
  flatten(array) {
    throw new Error("Remove this statement and implement this function");
  }
}
```

**Proposed Implementations**
1) Use constructor argument
```Javascript
// flatten-array.spec.js
import { Flattener } from './flatten-array.js';

describe('FlattenArray', () => {
  test('flattens a nested list', () => {
    const flattener = new Flattener([[]]);
    expect(flattener.flatten()).toEqual([]);
  });
}

// flatten-array.js
export class Flattener {
  constructor(array) {
    this.array = array;
  }
  flatten() {
    throw new Error("Remove this statement and implement this function");
  }
}
```

2) Use static method
```Javascript
// flatten-array.spec.js
import { Flattener } from './flatten-array.js';

describe('FlattenArray', () => {
  test('flattens a nested list', () => {
    expect(Flattener.flatten([[]])).toEqual([]);
  });
}

// flatten-array.js
export class Flattener {
  static flatten(array) {
    throw new Error("Remove this statement and implement this function");
  }
}
```

3) Use plain function
```Javascript
// flatten-array.spec.js
import { flatten } from './flatten-array.js';

describe('FlattenArray', () => {
  test('flattens a nested list', () => {
    expect(flatten([[]])).toEqual([]);
  });
}

// flatten-array.js
export function flatten() {
  throw new Error("Remove this statement and implement this function");
}
``` 

I would go with 3)

**Comment by SleeplessByte** [comment]
I think it currently makes sense, but I also _prefer_ your version three 😄.

@tejasbubane what do you think?

**Comment by gabriel376** [comment]
Regardless of what we decide, we should keep some consistency across exercises regarding naming, structure, etc
See [sieve](https://github.com/exercism/javascript/blob/master/exercises/sieve/sieve.js), for instance, that uses plain function.
Otherwise, if the exercise is about classes, makes sense to work with classes.

**Comment by tejasbubane** [comment]
> Regardless of what we decide, we should keep some consistency across exercises regarding naming, structure, etc
> See [sieve](https://github.com/exercism/javascript/blob/master/exercises/sieve/sieve.js), for instance, that uses plain function.
> Otherwise, if the exercise is about classes, makes sense to work with classes.

@gabriel376 We did have a discussion on this: https://github.com/exercism/javascript/issues/274

TLDR: For exercises which need initial state use classes otherwise use plain functions.

Although we did not do a thorough scrutiny for this rule on all exercises, contributors just chimed in and modified some exercises. This exercise seems to fit the functional style so I prefer the third option - `Use plain function` - `flatten([[]])`.

Would you like to submit a PR?

**Comment by gabriel376** [comment]
yep, I will submit a PR
thanks!
